### ISSUE TITLE
NHTSA database

### ISSUE THREAD

**Issue by tariqsameed** [SoftLegend/AC3R-Demo#1]
I want to know that AC3R covers the crash report of only NMVCSS database? Does your pre-crash event extraction algorithm work on the NASS database crash report.

**Comment by SoftLegend** [comment]
At the moment, AC3R cannot analyze NASS database because AC3R analyzes textual crash description in natural languages. Perhaps one can configure AC3R to decode and analyze crash data in NASS database for reconstructing accidents.

**Comment by alessiogambi** [comment]
What's the difference between the two databases? Would it be possible to adapt the current implementation? Why are we using different databases?

**Comment by tariqsameed** [comment]
As far as I know, after discussing with the peers who are working pre-crash event extraction, the NASS database contains more information and well written in textual crash description (natural language).  The processing of data becomes quite easier.

**Comment by alessiogambi** [comment]
So why can't we directly apply AC3R to it?

**Comment by SoftLegend** [comment]
AC3R can be modified to reconstruct NASS Database.

Currently, AC3R cannot analyze NASS cases because NASS does not use crash summary to describe the events and environment properties. It uses dictionary to map a code with a crash-related factor (crash position, pre-crash events, etc.). For example, to describe the action of a car before an impact, NASS CDS (Crashworthiness Data System) uses the following dictionary:

Code    Description
5           Stopped In Road
11         Turning Left
12         Turning Right

Similarly, weather, horizontal and vertical crash positions and other factors are assigned with specific codes. NASS GES (General Estimate System) also has its own dictionary to record crash related information.

In conclusion, one can add NASS GES or CDS dictionaries to AC3R, and implement functions to translate fundamental crash contributing factors for reconstructing the environment and crash events. Noticeably, for vehicle actions in the crash, each vehicle's sequence of action is coded **chronologically**, so parsing dependencies in sentence is no longer necessary since we know Vehicle 1 performed actions X, Y, Z in the crash.

**Comment by alessiogambi** [comment]
This sounds a great opportunity to implement a similar yet not the same systems to leverage police reports to construct simulations. The focus will be of course not on parsing and interpreting the text with nlp but in other aspects.
